nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,435 @@
+        .name("rest-lookup-ssl-context-service")
+        .displayName("SSL Context Service")
+        .description("The SSL Context Service used to provide client certificate information for TLS/SSL "
+                + "connections.")
+        .required(false)
+        .identifiesControllerService(SSLContextService.class)
+        .build();
+    public static final PropertyDescriptor PROP_BASIC_AUTH_USERNAME = new PropertyDescriptor.Builder()
+        .name("rest-lookup-basic-auth-username")
+        .displayName("Basic Authentication Username")
+        .description("The username to be used by the client to authenticate against the Remote URL.  Cannot include control characters (0-31), ':', or DEL (127).")
+        .required(false)
+        .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x39\\x3b-\\x7e\\x80-\\xff]+$")))
+        .build();
+
+    public static final PropertyDescriptor PROP_BASIC_AUTH_PASSWORD = new PropertyDescriptor.Builder()
+        .name("rest-lookup-basic-auth-password")
+        .displayName("Basic Authentication Password")
+        .description("The password to be used by the client to authenticate against the Remote URL.")
+        .required(false)
+        .sensitive(true)
+        .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x7e\\x80-\\xff]+$")))
+        .build();
+    public static final PropertyDescriptor PROP_DIGEST_AUTH = new PropertyDescriptor.Builder()
+        .name("rest-lookup-digest-auth")
+        .displayName("Use Digest Authentication")
+        .description("Whether to communicate with the website using Digest Authentication. 'Basic Authentication Username' and 'Basic Authentication Password' are used "
+                + "for authentication.")
+        .required(false)
+        .defaultValue("false")
+        .allowableValues("true", "false")
+        .build();
+
+    private static final ProxySpec[] PROXY_SPECS = {ProxySpec.HTTP_AUTH, ProxySpec.SOCKS};
+    public static final PropertyDescriptor PROXY_CONFIGURATION_SERVICE
+            = ProxyConfiguration.createProxyConfigPropertyDescriptor(true, PROXY_SPECS);
+
+    static final String MIME_TYPE_KEY = "mime.type";
+    static final String BODY_KEY = "request.body";
+    static final String METHOD_KEY = "request.method";
+
+    static final List<PropertyDescriptor> DESCRIPTORS;
+    static final Set<String> KEYS;
+
+    static {
+        DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            URL,
+            RECORD_READER,
+            RECORD_PATH,
+            RECORD_PATH_PROPERTY_NAME,
+            SSL_CONTEXT_SERVICE,
+            PROXY_CONFIGURATION_SERVICE,
+            PROP_BASIC_AUTH_USERNAME,
+            PROP_BASIC_AUTH_PASSWORD,
+            PROP_DIGEST_AUTH
+        ));
+        Set<String> _keys = new HashSet<>();
+        _keys.add(MIME_TYPE_KEY);
+        _keys.add(METHOD_KEY);
+        KEYS = Collections.unmodifiableSet(_keys);
```

Comment:
MIME_TYPE is only used for put and post requests and should be optional. Also if we support 'get' is the default method, then this controller service does not have any required KEYS.

Reply:
That's a fair point, and it needs validation on the verbs.

Reply:
Done.